src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+        /// </summary>
+        /// <param name="year">The year (1 through the number of years in calendar).</param>
+        /// <param name="month">The month (1 through the number of months in calendar).</param>
+        /// <param name="day">The day (1 through the number of days in month).<paramref name="month"/>.</param>
+        /// <param name="calendar">The calendar that is used to interpret year, month, and day.<paramref name="month"/>.</param>
+        public DateOnly(int year, int month, int day, Calendar calendar) => _dayNumber = DayNumberFromDateTime(new DateTime(year, month, day, calendar));
+
+        /// <summary>
+        /// Initializes a new instance of the DateOnly structure to the specified number of days.
+        /// </summary>
+        /// <param name="dayNumber">The number of days since January 1, 0001 in the Gregorian calendar.</param>
+        public static DateOnly FromDayNumber(int dayNumber)
+        {
+            if ((uint)dayNumber > MaxDayNumber)
+            {
+                ThrowHelper.ThrowArgumentOutOfRange_DayNumber(dayNumber);
+            }
+
+            return new DateOnly(dayNumber);
+        }
+
+        /// <summary>
+        /// Gets the year component of the date represented by this instance.
+        /// </summary>
+        public int Year => GetEquivalentDateTime().Year;
+
+        /// <summary>
+        /// Gets the month component of the date represented by this instance.
+        /// </summary>
+        public int Month  => GetEquivalentDateTime().Month;
+
+        /// <summary>
+        /// Gets the day component of the date represented by this instance.
+        /// </summary>
+        public int Day => GetEquivalentDateTime().Day;
+
+        /// <summary>
+        /// Gets the day of the week represented by this instance.
+        /// </summary>
+        public DayOfWeek DayOfWeek => GetEquivalentDateTime().DayOfWeek;
+
+        /// <summary>
+        /// Gets the day of the year represented by this instance.
+        /// </summary>
+        public int DayOfYear => GetEquivalentDateTime().DayOfYear;
+
+        /// <summary>
+        /// Gets the number if days since January 1, 0001 in the Gregorian calendar represented by this instance.
+        /// </summary>
+        public int DayNumber => _dayNumber;
+
+        /// <summary>
+        /// Returns a new DateOnly that adds the specified number of days to the value of this instance.
+        /// </summary>
+        /// <param name="value">A number of days. The value parameter can be negative or positive.</param>
+        /// <returns>An object whose value is the sum of the date represented by this instance and the number of days represented by value.</returns>
+        public DateOnly AddDays(int value)
+        {
+            int newDayNumber = _dayNumber + value;
+            if ((uint)newDayNumber > MaxDayNumber)
```

Comment:
Do you also have to check for less than minimum?

Reply:
uint check is doing the trick.